hedera-node/hedera-util-service-impl/src/main/java/com/hedera/node/app/service/util/impl/handlers/AtomicBatchHandler.java

```diff
@@ -93,7 +120,25 @@ public void pureChecks(@NonNull final PureChecksContext context) throws PreCheck
     @Override
     public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
-        // TODO
+        final var op = context.body();
+        final var atomicBatchTransactionBody = op.atomicBatchOrThrow();
+        requireNonNull(op);
+        List<Transaction> transactions = atomicBatchTransactionBody.transactions();
+
+        if (transactions.size()
+                > context.configuration().getConfigData(AtomicBatchConfig.class).maxNumberOfTransactions()) {
+            throw new PreCheckException(BATCH_SIZE_LIMIT_EXCEEDED);
```

Comment:
This check should be done in handle

Reply:
If you add in pre-handle they will be repeated in handle again. But I think adding in handle is what we did in other handlers